src/tests/functional/plugin/shared/src/behavior/ov_plugin/caching_tests.cpp

```diff
@@ -242,22 +242,34 @@ void CompileModelCacheTestBase::run() {
         GTEST_FAIL() << "Can't compile network without cache for " << m_functionName << " with precision " << m_precision.get_type_name() << std::endl;
     }
     auto originalOutputs = get_plugin_outputs();
-
+    size_t blobCountInitial = -1;
+    size_t blobCountAfterwards = -1;
     for (int i = 0; i < 2; i++) {
         // Step 2: Load with cache. Export or import shall not throw
         compiledModel = {}; // Destroy network object
         inferRequest = {};
         {
             core->set_property(ov::cache_dir(m_cacheFolderName));
             ASSERT_NO_THROW(compiledModel = core->compile_model(function, targetDevice, configuration));
-            if (targetDevice.find("AUTO") == std::string::npos)
-                // Apply check only for HW plugins
-                ASSERT_EQ(i != 0, compiledModel.get_property(ov::loaded_from_cache));
+            ASSERT_EQ(i != 0, compiledModel.get_property(ov::loaded_from_cache));
+            while (targetDevice.find("AUTO") != std::string::npos) {
+                auto exeDevices = compiledModel.get_property(ov::execution_devices);
+                if (exeDevices.size() == 1 && exeDevices.front().find("(CPU)") != std::string::npos)
```

Comment:
Please, remove hardcode -> `exeDevices.front().find("(CPU)")`

Reply:
@yangwang201911 seems some problem here, if GPU is compiling faster than cpu help, the logic below will fail.
let's put it this way, for AUTO, first iter : we just check the cache num >0,  at the very end , destroy the compile model to let auto finish async compiling, then record the cache num size
 second iter: compile and compare the result, destroy the compile model, then compare the cache num to the previously recorded one

Reply:
Updated.

Reply:
@songbell Updated.